README.md

```diff
@@ -22,49 +34,149 @@ You can do it manually following the instructions from [rust-lang.org](https://w
+
+## Lotus-CLI establishing
+### User approach:
+1. Download [lotus-CLI executable](https://github.com/filecoin-project/lotus/releases/ "lotus releases")
+2. You can use CLI client with your node by linking it with `--repo <path to your node working directory>` flag,
+   more of lotus-CLI flags will be covered in **Work with Lotus-CLI**
+
+### Advananced approach:
+1. clone [lotus repository](https://github.com/filecoin-project/lotus "lotus project")
+2. build lotus cli client with command: `make build`
+3. In case you want connect to interopnet, build lotus with `make interopnet`
+## Work with Lotus-CLI
+
+1. For proper miner work you will need a bls wallet, which can be made with lotus-cli by using the following command: `$lotus --repo <repo of your working node> wallet new bls`, after the execution of this comand you will revcieve the public address of your new account, and a private key in `<repo of youe working node>/keystore`.
+2. Also you may need to put some Filecoins on your account from other addresses you may have, and you can do it  with `lotus --repo <repo of your working node> send --from <sender address> <public address of your account>`
+
+3. If you dont have any Filecoins or you want to get more, you can buy them on stocks with usage of you public address.
+
+4. Other advanced commands you may find in [filecoin-docs about lotus CLI](https://docs.filecoin.io/mine/lotus/dagstore/#cli-commands "lotus CLI documentation").
+## Fuhon miner establishing
+To start mining at first you need to configure and run Fuhon, Lotus, Venus or any other filecoin node. Before all make sure that your node had successfully **connected** and updated to the current **block height** (actual network block number). Also, your node's api port should be opened for conection.
+
+To provide wallet functions and oracle information be sure to run lotus-cli at first. You can find how to do it in **Lotus-CLI establishing**.
 
-## Development
-### Build cpp-filecoin
+Once all the needed  modules will be started, you can continue with miner configuration.
 
-First build will likely take long time. However, you can cache binaries to [hunter-binary-cache](https://github.com/soramitsu/hunter-binary-cache) or even download binaries from the cache in case someone has already compiled project with the same compiler. To this end, you need to set up two environment variables:
+1. Create miner work directory (f.e. `$mkdir miner`)
+2. Download miner executable here : `https:\\link_to_download` and put it in work directory
+3. Download proof parameters `https:\\` and clone it in miner directory or if you have copy of the cpp-filecoin project, just copy proof parameters with command: `p /<path to cpp-filecoin>/core/proofs/parameters.json <your miner working directory>/proof-params.json`
+4. Run yor miner with ` --repo <node working directory> --miner-repo <miner work directory> --owner <your public address> --miner-api <node api open port> --sector-size <sector size>`
```

Comment:
```suggestion
4. Run yor miner with ` --repo <node working directory> --miner-repo <miner work directory> --owner <your public address> --miner-api <miner api port> --sector-size <sector size>`
```

another parameters can be specified to reuse existing miner actor.

Reply:
fixed

Reply:
MINER_API_PORT (not same as node API_PORT).

Reply:
done